### ISSUE TITLE
Unstable tests in Travis+Docker setup

### ISSUE THREAD

**Issue by olexandr-konovalov** [gap-packages/curlInterface#14]
I have restarted and tried several times. It sometimes passes and sometimes fails (e.g. https://travis-ci.org/gap-system/gap-docker-pkg-tests-stable-4.9/jobs/426072192) and it does not depend on the packages configuration:
```
testing: /home/gap/inst/gap-stable-4.9/pkg/curlInterface-2.0.2/tst/basic.tst
########> Diff in /home/gap/inst/gap-stable-4.9/pkg/curlInterface-2.0.2/tst/ba\
sic.tst:40
# Input is:
SortedList(RecNames(r));
# Expected output:
[ "result", "success" ]
# But found:
[ "error", "success" ]
########
########> Diff in /home/gap/inst/gap-stable-4.9/pkg/curlInterface-2.0.2/tst/ba\
sic.tst:42
# Input is:
r.success;
# Expected output:
true
# But found:
false
########
########> Diff in /home/gap/inst/gap-stable-4.9/pkg/curlInterface-2.0.2/tst/ba\
sic.tst:44
# Input is:
r.result = ListWithIdenticalEntries(1024, '\000');
# Expected output:
true
# But found:
Error, Record: '<rec>.result' must have an assigned value
########
```
What the test does is

    r := DownloadURL("ftp://speedtest.tele2.net/1KB.zip");; 

Perhaps there are intermittent problems with connectivity? Is there a way to say retry to download it five times before failing?

**Comment by mtorpey** [comment]
I can't reproduce this locally, but perhaps there are some clues as to what's going wrong.
* This is the only test that uses FTP.
* This is the only test that uses a server at `tele2.net`.
We don't actually advertise FTP functionality in the package documentation, so one option would be to remove the test, and only think about it again if we wanted to support FTP properly.

For the moment, I've left the test in but with a slightly different address that points specifically to a server in Frankfurt (commit 041b062).  Perhaps this might at least make the Docker tests pass or fail more consistently.  @alex-konovalov, I'd love to hear whether these continue to fail.

**Comment by olexandr-konovalov** [comment]
Thanks.
1. Please do not enclose commit id into backquotes - otherwise they are not rendered as links
2. Then the test will fail each time when the Frankfurt server is not available? Or that server also has some redundancy?
3. Let's see the outcome in the next run of tests at https://travis-ci.org/gap-packages/gap-docker-pkg-tests-master-devel - they use similar Travis+Docker setup but pull the latest _development_ version of the package being tested from its GitHub repository.

**Comment by mtorpey** [comment]
I've eventually managed to reproduce this using my own fork of Alex's docker tests.  I've got a verbose log of it working on my computer, and going wrong on the server.  This is the output from `DownloadURL("ftp://fra36-speedtest-1.tele2.net/1KB.zip", rec(verbose := true));`.  It starts the same on both computers (with port number `23873` changing each time):

```
* TCP_NODELAY set
* Connected to fra36-speedtest-1.tele2.net (90.130.74.155) port 21 (#0)
< 220 (vsFTPd 2.3.5)
> USER anonymous
< 331 Please specify the password.
> PASS ftp@example.com
< 230 Login successful.
> PWD
< 257 "/"
* Entry path is '/'
> EPSV
* Connect data stream passively
* ftp_perform ends with SECONDARY: 0
< 229 Entering Extended Passive Mode (|||23873|).
*   Trying 90.130.74.155...
* TCP_NODELAY set
* Connecting to 90.130.74.155 (90.130.74.155) port 23873
* Connected to fra36-speedtest-1.tele2.net (90.130.74.155) port 21 (#0)
> TYPE I
< 200 Switching to Binary mode.
> SIZE 1KB.zip
< 213 1024
> RETR 1KB.zip
```
At this point, the output diverges.  The successful test on my computer continues:
```
< 150 Opening BINARY mode data connection for 1KB.zip (1024 bytes).
* Maxdownload = -1
* Getting file with size: 1024
* Remembering we are in dir ""
< 226 Transfer complete.
* Connection #0 to host fra36-speedtest-1.tele2.net left intact
```
but on the server it instead goes:
```
< 425 Security: Bad IP connecting.
* RETR response: 425
* Remembering we are in dir ""
* Connection #0 to host fra36-speedtest-1.tele2.net left intact
```

I don't know much about "425: bad IP connecting", but I'm wondering if it has to do with the FTP connection somehow being blocked by docker.  Any thoughts, @alex-konovalov or @ChrisJefferson?

**Comment by olexandr-konovalov** [comment]
Ah, could be! Cf. https://github.com/gap-packages/gap-docker-pkg-tests-master-devel/issues/1

**Comment by mtorpey** [comment]
The offending test was removed in 59ff8dcf.  Could you verify whether this fixes it, @alex-konovalov?

**Comment by olexandr-konovalov** [comment]
@mtorpey seems so, thanks: https://travis-ci.org/gap-packages/gap-docker-pkg-tests-master-devel/jobs/445932466

**Comment by olexandr-konovalov** [comment]
@mtorpey will you be making a new release now? If today or tomorrow, it will be nice to have it in GAP 4.10!

**Comment by mtorpey** [comment]
I'll mark this as fixed, and release shortly!

**Comment by olexandr-konovalov** [comment]
Picked up - thanks!
```
* curlInterface - new version 2.1.1 discovered!!!
  ============================================
  Getting new archives from 
  https://github.com/gap-packages/curlInterface/releases/download/v2.1.1/curlI\
nterface-2.1.1[ ".tar.gz" ]
  unpacking curlInterface-2.1.1.tar.gz
  Validation of the info file successful!
  Package curlInterface 2.1.1 from 26/10/2018 has dependencies:
  * NeededOtherPackages [ [ "GAPDoc", ">= 1.5" ] ]
  * GAP >= 4.9
  finding text files  . . .

=====================text files==========================
curlInterface-2.1.1/README.md
curlInterface-2.1.1/VERSION
curlInterface-2.1.1/CHANGES
curlInterface-2.1.1/PackageInfo.g
curlInterface-2.1.1/init.g
curlInterface-2.1.1/GPL
curlInterface-2.1.1/makedoc.g
curlInterface-2.1.1/read.g
curlInterface-2.1.1/LICENSE
curlInterface-2.1.1/tst/basic.tst
curlInterface-2.1.1/tst/errors.tst
curlInterface-2.1.1/tst/testall.g
curlInterface-2.1.1/src/curl.c
curlInterface-2.1.1/gap/curl.gi
curlInterface-2.1.1/gap/curl.gd
curlInterface-2.1.1/doc/chap1_mj.html
curlInterface-2.1.1/doc/manual.css
curlInterface-2.1.1/doc/ragged.css
curlInterface-2.1.1/doc/chap1.txt
curlInterface-2.1.1/doc/lefttoc.css
curlInterface-2.1.1/doc/times.css
curlInterface-2.1.1/doc/chooser.html
curlInterface-2.1.1/doc/_Chapter_Overview.xml
curlInterface-2.1.1/doc/toggless.css
curlInterface-2.1.1/doc/_AutoDocMainFile.xml
curlInterface-2.1.1/doc/chap0.html
curlInterface-2.1.1/doc/title.xml
curlInterface-2.1.1/doc/curlInterface.xml
curlInterface-2.1.1/doc/chapInd.txt
curlInterface-2.1.1/doc/nocolorprompt.css
curlInterface-2.1.1/doc/chap0_mj.html
curlInterface-2.1.1/doc/chap1.html
curlInterface-2.1.1/doc/chapInd.html
curlInterface-2.1.1/doc/chap0.txt
curlInterface-2.1.1/doc/chapInd_mj.html

=====================end of the list of text files=======
=====================binary files========================
curlInterface-2.1.1/Makefile.am
curlInterface-2.1.1/Makefile.in
curlInterface-2.1.1/autogen.sh
curlInterface-2.1.1/configure.ac
curlInterface-2.1.1/configure
curlInterface-2.1.1/aclocal.m4
curlInterface-2.1.1/scripts/build_gap.sh
curlInterface-2.1.1/scripts/run_tests.sh
curlInterface-2.1.1/scripts/gather-coverage.sh
curlInterface-2.1.1/scripts/build_pkg.sh
curlInterface-2.1.1/cnf/compile
curlInterface-2.1.1/cnf/install-sh
curlInterface-2.1.1/cnf/depcomp
curlInterface-2.1.1/cnf/ltmain.sh
curlInterface-2.1.1/cnf/pkgconfig.h.in
curlInterface-2.1.1/cnf/config.guess
curlInterface-2.1.1/cnf/missing
curlInterface-2.1.1/cnf/ar-lib
curlInterface-2.1.1/cnf/config.sub
curlInterface-2.1.1/m4/libcurl.m4
curlInterface-2.1.1/m4/ltversion.m4
curlInterface-2.1.1/m4/ltsugar.m4
curlInterface-2.1.1/m4/ltoptions.m4
curlInterface-2.1.1/m4/libtool.m4
curlInterface-2.1.1/m4/ac_find_gap.m4
curlInterface-2.1.1/m4/lt~obsolete.m4
curlInterface-2.1.1/doc/manual.pdf
curlInterface-2.1.1/doc/manual.six
curlInterface-2.1.1/doc/toggless.js
curlInterface-2.1.1/doc/rainbow.js
curlInterface-2.1.1/doc/manual.js

=====================end of the list of binary files=====
```
